MetaMask/metamask-extension
# [TS dashboard] Draw dependencies between files

## Explanation

If you're thinking about converting a file to TypeScript, you might want
to know what files import that file and what files it imports. This
commit makes it so that if you click on a box in the visualization, it
will draw lines between that box and the other boxes which represent its
dependents and dependencies.

## Screenshots/Screencaps

https://user-images.githubusercontent.com/7371/220708405-07bc2fe7-656f-457a-8df7-622709581014.mov

## Manual Testing Steps

* Run `yarn ts-migration:dashboard:watch` in a terminal tab, then run open `development/ts-migration-dashboard/build/final/index.html`.
* Click on a box (let's call it A) that you see on the dashboard. You may see red lines and/or blue lines which connect this box to other boxes.
* Pick a red line and follow it up the tree to its box (let's call it B). This file is a dependency; if you open file A in your editor, you should see that it is `import`ing B.
* Pick a blue line and follow it down the tree to its box (let's call it C). This file is a dependent; if you open C in your editor, you should see that it is `import`ing A.
* Compare what you see to https://metamask.github.io/metamask-extension-ts-migration-dashboard/. While the boxes per level are now sorted, the boxes should live in the same levels — nothing should be rearranged.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

---

Closes #16810.
